src/microbe_stage/Patch.cs

```diff
@@ -107,6 +107,44 @@ public HashSet<Patch> GetAllConnectedPatches()
         return resultSet;
     }
 
+    /// <summary>
+    ///   Checks closest neighbours using breadth-first search (BFS) with the given maximum visits.
+    /// </summary>
+    /// <param name="visits">The maximum number of patches to visit/add</param>
+    /// <returns>A <see cref="HashSet{T}"/> of closest Patches connected to this node by some means</returns>
+    public HashSet<Patch> GetClosestConnectedPatches(int visits = 20)
+    {
+        var queue = new Queue<Patch>();
+        var visited = new HashSet<Patch>();
+
+        queue.Enqueue(this);
+        visited.Add(this);
+
+        var maxReached = false;
+
+        while (queue.Count > 0 && !maxReached)
+        {
+            var vertex = queue.Dequeue();
+
+            foreach (var patch in vertex.Adjacent)
+            {
+                if (!visited.Contains(patch))
+                {
+                    queue.Enqueue(patch);
+                    visited.Add(patch);
```

Comment:
I think `Add` already returns a boolean indicating if the value was new or not, so this could be used in the if on line 131.